src/pudl_archiver/archivers/msha.py

```diff
@@ -0,0 +1,124 @@
+"""Download MSHA data."""
+
+import logging
+import re
+from pathlib import Path
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+URL_BASE = "https://arlweb.msha.gov/OpenGovernmentData/"
+EXT_BASE = "OGIMSHA.asp"
+
+URL_107A = "https://arlweb.msha.gov/OpenGovernmentData/107a/"
+EXT_107A = "107aOrders.asp"
+
+datasets = {
```

Comment:
We try to follow the python convention that all module level constants use `SCREAMING_CASE`. Maybe make it a bit less generic for readability? `MSHA_DATASETS`?